pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -209,7 +218,7 @@ func (s *Service) handle(msg *message) error {
 		return fmt.Errorf("invalid state name: %w", err)
 	}
 
-	followup, err := next.Execute(msg.Msg)
+	followup, err := next.Execute(msg.Msg, s.ctx)
```

Comment:
instead of custom context, we can pass provider here

Reply:
if the context was intended to prevent another dependency in the future we can use `options`